### ISSUE TITLE
EDF+ Annotation Timestamps missing submillisecond accuracy

### ISSUE THREAD

**Issue by greydongilmore** [mne-tools/mne-python#7872]
It was noticed that the onset time for annotations in an EDF+ file might be missing the submillisecond accuracy accociated with the start time of the recording. This results in annotation timestamps being delayed from the true event time. 

#### Expected results

The EDF+ file used here contains patient data and cannot be shared. However, if this bug is confirmed all EDF+ files will be affected. 

If the EDF+ file is brought into [EDFbrowser](https://www.teuniz.net/edfbrowser/) it will be noticed that the millisecond aspect of the start time is displayed in the `Start` field:

![thumbnail_Outlook-ibjnd242](https://user-images.githubusercontent.com/26314366/83948860-4c288380-a7ee-11ea-8f98-9f95026da92d.png)

For instance, this results in the annotation timestamps being:

![thumbnail_Outlook-of2qriv1](https://user-images.githubusercontent.com/26314366/83948874-68c4bb80-a7ee-11ea-8e9f-6cd5969134cf.png)

#### Actual results

When loading the data into MNE the annotation timestamps return:

![thumbnail_Outlook-dnk5ugfi](https://user-images.githubusercontent.com/26314366/83948890-8abe3e00-a7ee-11ea-865b-d45f36dbff99.png)

If you take the difference between the MNE timestamp and the EDFbrowser timestamp it is equal to the submillisecond portion of the recording start time.

#### Additional information

I reached out to Teunis van Beelen, the developer of [EDFbrowser](https://www.teuniz.net/edfbrowser/), to get an explanation. The explanation provided was as follows:

> EDF does not have support for subsecond precision of the starttime but EDF+ does. Because EDF+ needed to be backwards compatible with EDF and EDF doesn't have the possibility to write subsecond precision for the starttime into the header, the subsecond part of the starttime is written outside of the EDF header. The way the subsecond part of the starttime is stored in EDF+ is described here: https://www.edfplus.info/specs/edfplus.html#timekeeping

Teunis van Beelen provided a quote from the EDF specs:

> "The startdate/time of a file is specified in the EDF+ header fields 'startdate of recording' and 'starttime of recording'. These fields must indicate the absolute second in which the start of the first data record falls. So, the first TAL in the first data record always starts with +0.X2020, indicating that the first data record starts a fraction, X, of a second after the startdate/time that is specified in the EDF+ header. If X=0, then the .X may be omitted."

In other words, the subsecond part of the starttime is written in the EDF Annotations "signal".

**Comment by agramfort** [comment]
do you know any public file that would allow us to
replicate/investigate this?

otherwise you can try to hack the edf.py file. Relevant code starts here:

https://github.com/mne-tools/mne-python/blob/master/mne/io/edf/edf.py#L231

**Comment by greydongilmore** [comment]
I have de-identified the file I used:
https://drive.google.com/file/d/1yql0bMGcXXrEZr1At_Cp-eMi6rTKSr_F/view?usp=sharing

**Comment by agramfort** [comment]
can you try this PR https://github.com/mne-tools/mne-python/pull/7875

your explanations helped a lot !

>

**Comment by greydongilmore** [comment]
Glad I could contribute. The explanations provided by Teunis van Beelen helped me to understand the issue so credit should be given to him.

**Comment by greydongilmore** [comment]
@agramfort @cbrnr I tested the response of FieldTrip and I am getting the same behavior. Seems to be missing the submillisecond accuracy:
![Selection_231](https://user-images.githubusercontent.com/26314366/84077193-f0ecc180-a9a4-11ea-8139-cfb3e16722ea.png)

Will reference this PR on their repo

**Comment by cbrnr** [comment]
Biosig also doesn't support this, I've reported this to the maintainer. Seems like edflib is the only app with proper support (but now also MNE).

**Comment by greydongilmore** [comment]
@cbrnr interestingly though [pyEDFlib](https://github.com/holgern/pyedflib/issues/79) is a python wrapper for EDFlib and it also shows the incorrect behavior... just tested

**Comment by skjerns** [comment]
@greydongilmore this is because we haven't integrated the new EDFlib version 1.17 yet (see https://github.com/holgern/pyedflib/pull/81)

I hope I find time soon to implement the changes.
